### PR TITLE
Refactor sigelt -> sigelt' (inductive) + sigelt (record w/ range and docs)

### PR DESCRIPTION
**Pull request by cpitclaudel**
Just like #917, but it builds! Ahem 😳

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Rebased on latest master to fix conflict in OCaml sources.

**Comment by cpitclaudel** [comment]
Rebased on latest master (no conflicts this time)

**Comment by cpitclaudel** [comment]
Rebased on latest master (conflicts in Tc.fs due to Aseem's refactoring)

File: src/syntax/FStar.Syntax.Syntax.fs [hunk 1]
```diff
@@ -33,6 +33,16 @@ type withinfo_t<'a,'t> = {
   p: Range.range;
 }
 
+// Documentation comment. May appear appear as follows:
```
Comment (msprotz) [inline src/syntax/FStar.Syntax.Syntax.fs]: nit

**Comment by msprotz** [comment]
Congratulations, this had been bothering me for a long time and I hadn't had the courage to tackle this unrewarding refactoring/cleanup. Thanks!

**Comment by cpitclaudel** [comment]
Yay, thanks!

**Pull request merged**